packages/dashboard/src/msw/iot-sitewise/resources/assets/index.ts

```diff
@@ -27,6 +27,12 @@ class AssetHierarchyClient {
     return rootAssets;
   }
 
+  public getAssetsByAssetModelId(assetModelId: string): Asset[] {
+    const assetList: Asset[] = [];
+    this.#searchByAssetModelId(assetModelId, this.#assetHierarchy, assetList);
```

Comment:
Try to avoid creating APIs which mutate references. Can you make this immutable? For example, have `#searchByAssetModelId` return a list of assets.